daemon/mgr/container.go

```diff
@@ -1028,6 +1050,42 @@ func (mgr *ContainerManager) parseVolumes(ctx context.Context, c *types.Containe
 	return nil
 }
 
+func (mgr *ContainerManager) detachVolumes(ctx context.Context, c *ContainerMeta) error {
+	for name := range c.Config.Volumes {
+		v, err := mgr.VolumeMgr.Get(ctx, name)
+		if err != nil {
+			logrus.Errorf("failed to get volume: %s", name)
+			return err
+		}
+
+		option := map[string]string{}
+		ref := v.Option("ref")
+		if ref == "" {
+			continue
+		}
+		if !strings.Contains(ref, c.ID) {
+			continue
+		}
+
+		newRef := []string{}
+		ids := strings.Split(ref, ",")
+		for _, id := range ids {
+			if id != c.ID {
+				newRef = append(newRef, id)
+			}
+		}
```

Comment:
Could we wrap a function like this:
```
func remove(s []string, r string) []string {
    for i, v := range s {
        if v == r {
            return append(s[:i], s[i+1:]...)
        }
    }
    return s
}
```
Could refer: https://stackoverflow.com/questions/34070369/removing-a-string-from-a-slice-in-go